plugins/modules/s3_metrics_configuration.py

```diff
@@ -0,0 +1,246 @@
+        if payload['Filter']['And'].get('Prefix') is not None:
+            parsed['filter_prefix'] = payload['Filter']['And']['Prefix']
+
+    if parsed.get('mc_id') != mc_id:
+        return False
+    if parsed.get('filter_prefix') != filter_prefix:
+        return False
+    if parsed.get('filter_tags', {}) != filter_tags:
+        return False
+    return True
+
+
+def create_or_update_metrics_configuration(client, module):
+    changed = False
+    bucket_name = module.params.get('bucket_name')
+    mc_id = module.params.get('id')
+    filter_prefix = module.params.get('filter_prefix')
+    filter_tags = module.params.get('filter_tags')
+
+    try:
+        metrics_configuration = client.get_bucket_metrics_configuration(Bucket=bucket_name, Id=mc_id)
+    except is_boto3_error_code('NoSuchConfiguration'):
+        metrics_configuration = None
+    except ClientError as e:  # pylint: disable=duplicate-except
+        module.fail_json_aws(e, msg="Failed to get bucket metrics configuration")
+
+    if metrics_configuration:
+        if not _compare_metrics_configuration(metrics_configuration, mc_id, filter_prefix, filter_tags):
+            try:
+                if not module.check_mode:
+                    client.put_bucket_metrics_configuration(
+                        Bucket=bucket_name,
+                        Id=mc_id,
+                        MetricsConfiguration=_create_metrics_configuration(mc_id, filter_prefix, filter_tags)
+                    )
+                changed = True
+            except (BotoCoreError, ClientError) as e:
+                module.fail_json_aws(e, msg="Failed to udpate bucket metrics configuration '%s'" % mc_id)
+    else:
+        try:
+            if not module.check_mode:
+                client.put_bucket_metrics_configuration(
+                    Bucket=bucket_name,
+                    Id=mc_id,
+                    MetricsConfiguration=_create_metrics_configuration(mc_id, filter_prefix, filter_tags)
+                )
+            changed = True
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json_aws(e, msg="Failed to create bucket metrics configuration '%s'" % mc_id)
+
+    module.exit_json(changed=changed)
+
+
+def delete_metrics_configuration(client, module):
+    changed = False
+    bucket_name = module.params.get('bucket_name')
+    mc_id = module.params.get('id')
+
+    try:
+        client.get_bucket_metrics_configuration(Bucket=bucket_name, Id=mc_id)
+    except is_boto3_error_code('NoSuchConfiguration'):
+        module.exit_json(changed=changed)
+    except ClientError as e:  # pylint: disable=duplicate-except
+        module.fail_json_aws(e, msg="Failed to get bucket metrics configuration")
+
+    try:
+        if not module.check_mode:
+            client.delete_bucket_metrics_configuration(Bucket=bucket_name, Id=mc_id)
+        changed = True
+    except (BotoCoreError, ClientError) as e:
```

Comment:
```suggestion
    except is_boto3_error_code('NoSuchConfiguration'):
        module.exit_json(changed=changed)
    except (BotoCoreError, ClientError) as e:  # pylint: disable=duplicate-except
```

Again, it's an edge-case but there's a chance that a delete request was in flight and completed between the Get and Delete operations.

We see things like this much more with S3 based services, so while I generally wouldn't try to catch this edge case, because this is S3, I would.